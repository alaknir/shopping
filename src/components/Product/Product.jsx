import React from "react";

import ProductSpeciality from "../ProductSpeciality";

const Product = ({ imgSrc, productName, isSale, isExclusive, price }) => (
  <div className="product">
    <img src={imgSrc} alt={productName} className="product__product-img" />
    <ProductSpeciality isSale={isSale} isExclusive={isExclusive} />
    <div>
      <span className="product__product-name">{productName}</span>
      <span className="product__product-price">{price}</span>
    </div>
  </div>
);

export default Product;
