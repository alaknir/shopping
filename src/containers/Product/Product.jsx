import React from "react";
import { withRouter, Redirect } from "react-router-dom";
import { connect } from "react-redux";

const Product = ({ products, match }) => {
  if (!products || !products.length) {
    return <Redirect to="/products" />;
  }

  const product = products.filter(product => match.params.id === product.id);
  const {
    isExclusive,
    isSale,
    price,
    productImage,
    productName,
    size
  } = product[0];
  return (
    <div>
      <img src={`../assets/products/${productImage}`} alt={productName} />
      <div>{isSale}</div>
      <div>{isExclusive}</div>
      <div>
        <span>{productName}</span>
        <span>{price}</span>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  products: state.products.data
});

export default connect(mapStateToProps)(withRouter(Product));
