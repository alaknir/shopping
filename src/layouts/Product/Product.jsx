import React from "react";
import ProductContainer from "../../containers/Product";

const Products = () => <ProductContainer />;

export default Products;
