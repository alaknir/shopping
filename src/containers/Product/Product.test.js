import React from "react";
import { mount } from "enzyme";

import Product from "./Product";

describe("Product container", () => {
  it("Should render properly", () => {
    const component = <Product />;
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
