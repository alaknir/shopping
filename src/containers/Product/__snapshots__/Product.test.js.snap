// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product container Should render properly 1`] = `
<Product>
  <div>
    Product
  </div>
</Product>
`;
